src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_raster/LocalServerDynamicWorkspaceRaster.java

```diff
@@ -0,0 +1,204 @@
+	        addButton.setMaxWidth(Double.MAX_VALUE);
+	        addButton.setDisable(true);
+
+
+	        // choose the file, then start the Local Server instance and the local map service...
+	        addButton.setOnAction(e -> {
+	          openFileChooser(stage);        
+	        });
+
+	        
+
+	        // add button to the control panel
+	        vBoxControl.getChildren().addAll(description, addButton);
+
+	        ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+	      
+	        map.addDoneLoadingListener(() -> addButton.setDisable(false));
+
+	        mapView = new MapView();
+	        mapView.setMap(map);
+
+	        // add the map view and control panel to stack pane
+	        stackPane.getChildren().addAll(mapView, vBoxControl);
+	        StackPane.setAlignment(vBoxControl, Pos.TOP_LEFT);
+	        StackPane.setMargin(vBoxControl, new Insets(10, 0, 0, 10));
+
+	      } catch (Exception e) {
+	        // on any error, display the stack trace
+	        e.printStackTrace();
+	      }
+		
+	}
+	
+	private void openFileChooser(Stage mainStage) {
+		
+		// Browse to the raster file
+		 FileChooser fileChooser = new FileChooser();
+		 fileChooser.setTitle("Open Resource File");
+		 fileChooser.getExtensionFilters().addAll(
+		         new ExtensionFilter("Image Files", "*.tif"));
+		 
+		 fileChooser.setInitialDirectory(new File("./samples-data/raster/"));
+		 File selectedFile = fileChooser.showOpenDialog(mainStage);
+		 
+		 if(selectedFile != null){
+		 
+			 String fileName = selectedFile.getName();
+			 String path = selectedFile.getParent();
+		 
+			 startLocalService(fileName, path);
+		 }
+		 
+		
+	}
+
+	private void startLocalService(String fileName, String path) {
+		
+		 // start local server
+	      server.startAsync();
+	      server.addStatusChangedListener(status -> {
+	        if (server.getStatus() == LocalServerStatus.STARTED) {
+	        	
+	          // start a service from the blank MPK...
+	          String mapServiceURL = "./samples-data/local_server/mpk_blank.mpk";
+	          localMapService = new LocalMapService(mapServiceURL);
+	          
+	          // Can't add a dynamic workspace to a running service, so do that first...
+	          RasterWorkspace rasterWS = new RasterWorkspace("raster_wkspc", path);
+	          RasterSublayerSource source = new RasterSublayerSource(rasterWS.getId(),fileName);
+	          newSL = new ArcGISMapImageSublayer(101, source);
+	          Iterable<DynamicWorkspace> dynamicWorkspaces = Arrays.asList(rasterWS);
+	          localMapService.setDynamicWorkspaces(dynamicWorkspaces);
+	          
+	          
+	          localMapService.addStatusChangedListener(new LocalService.StatusChangedListener(){
+
+				@Override
+				public void statusChanged(LocalService.StatusChangedEvent event) {
+					if (event.getNewStatus() == LocalServerStatus.STARTED){
+						
+						// Now, we're ready to add the raster layer...
+						addLocalMapImageLayer();
```

Comment:
This doesn't need to be in its own method. You can put the logic inside this lambda. It also eliminates the need to make `localMapService` and `newSL` class members.